### DISCUSSION TITLE
Fresh Apple M1 install of homebrew and python, need to update path?

### DISCUSSION THREAD

**Discussion by TXBDan** [Homebrew/discussions#476] [Getting started] [answered]
Sorry for the noob question. I just installed homebrew (arm native install) with no issues. I then installed python via "brew install python" (3.9.1) and again no issues. I see the bins are hidden away here: /opt/homebrew/opt/python3/bin

Do I simply add this path to my PATH to be able to access them? Just want to double check in case its better to link them or something. Some text during the python install mentioned linking, but Im not clear on if I was a warning or an action.

Thanks

**Comment by Rylan12** [comment, marked as answer]
Are you having trouble using python? If you've added `/opt/homebrew/bin` to your path (which seems likely given that you were able to run `brew install python`) you should be good. Homebrew automatically links binaries to `$(brew --prefix)/bin` which is `/opt/homebrew/bin` on arm macs.

Check `which -a python3` to make sure that `/opt/homebrew/bin/python3` shows up. If so, `python3 -V` should return 3.9.1.

If not, check `brew doctor`. If it has a warning saying you `python@3.9` us unlinked, running `brew link python@3.9` should solve your problem.

**Reply by TXBDan** [reply]
oh I see that works. I was confused because I thought it would install python 2 and 3 and I thought "python" would link to "python3". I saw that "python" was going to the default Mac python 2 install and thought it was broken. I'll just use "python3". Thanks

**Reply by Rylan12** [reply]
From `brew info python`:

```
Unversioned symlinks `python`, `python-config`, `pip` etc. pointing to
`python3`, `python3-config`, `pip3` etc., respectively, have been installed into
  /opt/homebrew/opt/python@3.9/libexec/bin
```

If you want unversioned symlinks (meaning `python` points to Python 3), you can add `/opt/homebrew/opt/python@3.9/libexec/bin` to your path.

It's possible that this will have unexpected consequences, though, if a command you're running expects `python` to point to be version 2.

**Reply by gogohaku** [reply]
Hi Rylan12,

Would you do me a favor?
I am also using MBA M1 and having difficulty to change default pyhon3 as brew version.
python3 was installed via brew and typed couple of command like "**ln -s python3 python**" to change brew version python3 as a default command.
In the end, I found that I made some mistake that I can't solve.
Do you know how to set **/opt/homebrew/bin/python3** as python3 command?
Attached command line on my Mac for more information.

> tommy@MacBook-Air bin % python3 -V               
> Python 3.8.2
> tommy@MacBook-Air bin % which -a python3
> /opt/homebrew/bin/python3
> /usr/bin/python3
> tommy@MacBook-Air bin % type python3             
> python3 is /usr/bin/python3

**Reply by Rylan12** [reply]
Hmm, not sure. Maybe try `brew reinstall python`?

I guess it's also possible that the `ln -s` command caused trouble. Where did you run that command?

You could also double-check your `PATH` to make sure that `/opt/homebrew/bin` is before `/usr/bin`, although my guess is that this is already done.

**Reply by gogohaku** [reply]
Thank you for your reply.
I typed `ln- s` at `/opt/homebrew/bin `.
Yes. as you guessed, when I checked `PATH` `/opt/homebrew/bin` is before `/usr/bin`
I am in trouble. hmmmm

**Reply by Rylan12** [reply]
What does `ls -l /opt/homebrew/bin/python*` show?

**Reply by gogohaku** [reply]
```
lrwxr-xr-x  1 tommy  admin  40 Mar 15 01:08 /opt/homebrew/bin/python3 -> ../Cellar/python@3.9/3.9.2_1/bin/python3
lrwxr-xr-x  1 tommy  admin  47 Mar 15 01:08 /opt/homebrew/bin/python3-config -> ../Cellar/python@3.9/3.9.2_1/bin/python3-config
lrwxr-xr-x  1 tommy  admin  42 Mar 15 01:08 /opt/homebrew/bin/python3.9 -> ../Cellar/python@3.9/3.9.2_1/bin/python3.9
lrwxr-xr-x  1 tommy  admin  49 Mar 15 01:08 /opt/homebrew/bin/python3.9-config -> ../Cellar/python@3.9/3.9.2_1/bin/python3.9-config
```

It shows like this.
Thank you.

**Reply by gromgit** [reply]
> ```bash
> tommy@MacBook-Air bin % python3 -V
> Python 3.8.2
> tommy@MacBook-Air bin % which -a python3
> /opt/homebrew/bin/python3
> /usr/bin/python3
> tommy@MacBook-Air bin % type python3
> python3 is /usr/bin/python3
> ```

Looks like you ran the system Python before installing Homebrew Python, so the former is still in zsh's command hash table. Run the following:
```bash
hash -r
type python3
```
That last command should then print `python3 is /opt/homebrew/bin/python3`.

**Reply by gogohaku** [reply]
Hi, gromgit!

OMG! `type python3` returned `/opt/homebrew/bin/python3`.
It's cleared!

Thank you so much!!

**Reply by emil-ep** [reply]
Ran into the same issue mentioned by @gogohaku . The solution mentioned by @gromgit  solved the issue,. Thanks !

**Reply by amathamsetty** [reply]
Hi @Rylan12  How can we change the python path to /usr/local/bin because brew installs it to /opt/homebrew/bin/python3 ? I have tried changing the .zprofile but did not help. Can you help me with this?

**Reply by prasanna-munnangi** [reply]
@amathamsetty , with apple silicon the default location of brew has been changed to /opt/homebrew from /usr/local/bin. Usually it is not recommended to change the default location
